### DISCUSSION TITLE
multi account support

### DISCUSSION THREAD

**Discussion by nycex** [owntracks/android#1072] [Ideas & New Features]
It would be a really nice addition to OwnTracks to have the ability to log in with multiple accounts. Those could then for instance be enabled/disabled separately, e.g. just like in Conversations as a XMPP client.
One exemplary use case for this would be to use OwnTracks on your own privately and going to an event which has some kind of use for OwnTracks where they give everyone an account on their MQTT server. Then, if the event is recurring, you could just disable the account for the time being and enable it again when the event takes places once more.

**Comment by jpmens** [comment]
Thanks for the suggestion, but I think this is something we are not likely to be doing in any short-term future, if only because it's very unlikely people will start adding friends on and off on their brokers.

BTW if you do want to now already do something along these lines you can using our [Remoteconfig](https://owntracks.org/booklet/features/remoteconfig/). Pre-configure settings which you load from file or even by clicking on a URL and switch back to other settings by opening a second file (or clicking on a 2nd URL).

**Comment by growse** [comment]
I think I'm with @jpmens  here - this would bring a fair bit of complexity to solve a problem most people don't have.

> One exemplary use case for this would be to use OwnTracks on your own privately and going to an event which has some kind of use for OwnTracks where they give everyone an account on their MQTT server. Then, if the event is recurring, you could just disable the account for the time being and enable it again when the event takes places once more.

Would a better solution here not be to just bridge your personal broker to the event's broker for however long you wanted it for?

**Comment by nycex** [comment]
Well there are 2 problems I can see with that approach:
- Not everyone has his own broker, it can be either hosted somewhere with certain rules where you can share your data with other accounts on that host or you could just have a shared one with for instance your family.
- Sharing a broker would - in this use case - complicate everything since if you could just add another account, as the host of that event, you could just create a bunch of account and give it to all participants, if they already have a MQTT account or not. Also, sharing a broker cannot be really done if you are not at home, since you have to do it on the server and you have to have the event's broker's information beforehand. It would create a organisational mess.

I also don't really agree with the notion that most people don't have this problem, since without, some use cases where OwnTracks otherwise might be applicable just can't really use it right now, but they could if OwnTracks had this feature.

**Comment by jpmens** [comment]
We're confident that most people don't have this problem.

**Comment by growse** [comment]
> I also don't really agree with the notion that most people don't have this problem, since without, some use cases where OwnTracks otherwise might be applicable just can't really use it right now, but they could if OwnTracks had this feature.

This is tautalogical. Yes, if users could do something, then they could do something. But that's not the same thing as actually solving a real problem a significant number of people have.

That said, we have no data. Maybe there's loads of people who need this - I'm going to move this to a discussion and let people vote if they want to see if implemented. If so, we can revisit.

**Comment by iarp** [comment]
I'd like to +1 this idea. I believe he encryption key should be in with the connection details as that would be server specific rather than a global setting.
